### ISSUE TITLE
Powergrids library coverage

### ISSUE THREAD

**Issue by max-privato** [OpenModelica/OpenModelica#9030]
### Description
From this link:
https://libraries.openmodelica.org/branches/overview.html
I read that OM 1.18.0 should be able to run all the PowerGrids 1.0.2 models.
However this happens not to be the case, at least for model 
Examples.Tutorial.IslandOperation.TwoGeneratorsLocalInitialization

### Steps to Reproduce
1. download and unzip PowerGrids 1.0.2 zip file
2. load PowerGrids onto OMEdit and run Examples.Tutorial.IslandOperation.TwoGeneratorsLocalInitialization
3. The following error pops up:

![image](https://user-images.githubusercontent.com/37046832/170835120-84fcc56a-6c6c-4d1a-af30-8c9b629fb15b.png)

### Expected Behavior
The model should run successfully

### Version and OS
OM 1.18.0 Win 10 64 bit

 - OpenModelica Version: [e.g. `omc --version` or `Help->About OMEdit` from OMEdit]
 - OS: [e.g. Windows 10, 64 bit]
 - Versions of used Modelica libraries if applicable

**Comment by casella** [comment]
@max-privato this is a known issue, see PowerGrids/PowerGrids#40. I'm still not sure if the model is structurally bad, or if it is just OpenModelica that is not good enough to solve it. 

Have you tried running the same model with Dymola?

**Comment by max-privato** [comment]
I just wanted to show inconsistency between the coverage report and what rrally happens, not the fact that this model is not run.
Dymola is not able to run any of the three transient Tutorial.IslandOperation models, but this is another story.

**Comment by bilderbuchi** [comment]
AFAICT, this model is not included in the coverage testing at all. I [noticed](https://github.com/PowerGrids/PowerGrids/blob/master/PowerGrids/Examples/Tutorial/IslandOperation/TwoGeneratorsLocalInitialization.mo) that the model does not have an `experiment` annotation, the presence of which I think OM testing uses to pick up models to run. 
Note that the model it *extends from* does have that annotations, but I don't know off the cuff if an experiment annotation is preserved during inheritance?

**Comment by perost** [comment]
> AFAICT, this model is [not included in the coverage testing ](https://libraries.openmodelica.org/branches/v1.18/PowerGrids/PowerGrids.html)at all. I [noticed](https://github.com/PowerGrids/PowerGrids/blob/master/PowerGrids/Examples/Tutorial/IslandOperation/TwoGeneratorsLocalInitialization.mo) that the model does not have an `experiment` annotation, the presence of which I think OM testing uses to pick up models to run. Note that the model it _extends from_ does have that annotations, but I don't know off the cuff if an experiment annotation is preserved during inheritance?

We do indeed use the `experiment` annotation to decide which models to test, but we only check the annotations directly in a model. How to test libraries is not currently part of the specification, so that's just how we've chosen to do it.

It would be considerably more expensive and complicated to have to look up every base class of every class in a library to check if they have `experiment` annotations, since this involves at least partial instantiation rather than just scanning the abstract syntax tree. It's also not certain that an "inherited" `experiment` annotation would be appropriate for the extended class, and it's also not specified how that's supposed to work (what if there are multiple `experiment` annotations?).

There's some discussion about this in [#2999](https://github.com/modelica/ModelicaSpecification/pull/2999) for the specification though, so maybe we'll change this in the future.

**Comment by bilderbuchi** [comment]
> It's also not certain that an "inherited" experiment annotation would be appropriate for the extended class...

I agree, that's also something I was wondering about.

**Comment by casella** [comment]
In fact, the problem I mentioned in PowerGrids/PowerGrids/#40 is not the same that @max-privato mentioned, it is rather a regression, since the model simulated successfully in 1.18.0, though I'm not 100% sure that the result was correct. This is also something we should look into, if we had enough time 😅

**Comment by perost** [comment]
> As @perost wrote, every model needs to have its own experiment annotations, using inheritance for it would be complicated, also because you may have to merge them somehow if there are different experiment annotations in the inheritance tree. If I am not mistaken, the Modelica Specification only mentions inheritance in annotations w.r.t. the graphical icon annotation, where the obvious idea is to draw the graphical annotations of children on top of those of their ancestors.

There's some discussion in the ticket I mentioned above for allowing non-simulation models to have `experiment` annotations in order to provide e.g. recommended settings, so in that case it would be inherited. But I think in that case an extending model should still have its own `experiment` annotation, even if it's empty, just to mark that the intent is indeed to use it for testing.

**Comment by casella** [comment]
So I guess the bottom line is:
- we should make sure that the TwoGeneratorsLocalInitialization model works fine, see #9033 
- once that is done, we should add an experiment annotation to it, so that it is added to the library testsuite

Then we can close this ticket 😄
